crates/metadata-providers/src/objstore/version_repository.rs

```diff
@@ -42,9 +47,44 @@ impl Tag {
     }
 }
 
+#[derive(Debug, Clone, thiserror::Error)]
+pub enum EncodingError {
+    #[error("Unknown encoding '{0}'")]
+    UnknownEncoding(String),
+}
+
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
+pub(crate) enum ValueEncoding {
+    #[default]
+    Ciborium,
+    Bilrost,
+}
+
+impl FromStr for ValueEncoding {
+    type Err = EncodingError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "binary/bilrost" => Ok(Self::Bilrost),
+            "binary/ciborium" => Ok(Self::Ciborium),
+            // default for backward compatibility
+            "" => Ok(Self::Ciborium),
```

Comment:
The default is probably unset so we'll just fall back to the `Default` impl, right? Maybe more explicit bail for empty attributes too since we've never set this object attribute before. IDK, probably doesn't hurt.